File: ACTIVIDAD_INTERMEDIA_MODULO_2/validar_telefonos/validar_telefono_espana.py
import re

def validar_telefono_espana(telefono: str) -> bool:
    """
    Valida si un número de teléfono sigue el formato internacional español.
    
    El formato requerido es: '+34' seguido exactamente de 9 dígitos numéricos.
    
    Args:
        telefono (str): La cadena de texto con el número a validar.
        
    Returns:
        bool: True si el formato es válido, False en caso contrario.
    """
    # Explicación del patrón:
    # ^      : Inicio de la cadena
    # \+34   : El literal '+34' (escapamos el + porque es un caracter especial)
    # [6789] : El primer dígito suele empezar por 6, 7 (móviles) o 8, 9 (fijos)
    # \d{8}  : Exactamente 8 dígitos numéricos adicionales
    # $      : Fin de la cadena
    patron = r"^\+34[6789]\d{8}$"
    
    return bool(re.match(patron, telefono))

# --- Casos de Prueba ---
pruebas = [
    "+34600111222",  # Válido (Móvil)
    "+34912345678",  # Válido (Fijo)
    "600111222",     # Inválido (Falta el prefijo +34)
    "+34123456789",  # Inválido (Empieza por 1, no es un rango estándar)
    "+346001112223", # Inválido (Tiene 10 dígitos en lugar de 9)
    "34600111222",   # Inválido (Falta el símbolo +)
]

print("Resultados de la validación:")
for t in pruebas:
    resultado = "✅ Válido" if validar_telefono_espana(t) else "❌ Inválido"
    print(f"{t}: {resultado}")

File: ACTIVIDAD_INTERMEDIA_MODULO_2/validar_telefonos/validarTelefonoEspana.js
/**
 * Valida si un número de teléfono sigue el formato internacional español (+34XXXXXXXXX).
 * * @param {string} telefono - El número de teléfono a validar.
 * @returns {boolean} - Retorna true si es válido, false de lo contrario.
 */
function validarTelefonoEspana(telefono) {
    // Explicación del patrón:
    // ^        : Inicio de la cadena.
    // \+34     : El prefijo literal +34.
    // [6789]   : El primer dígito tras el prefijo (6, 7 para móvil; 8, 9 para fijo).
    // \d{8}    : Exactamente 8 dígitos numéricos adicionales.
    // $        : Fin de la cadena.
    const patron = /^\+34[6789]\d{8}$/;

    return patron.test(telefono);
}

// --- Casos de Prueba ---
const pruebas = [
    { num: "+34611222333", esperado: true },  // Móvil válido
    { num: "+34932112233", esperado: true },  // Fijo válido
    { num: "611222333",    esperado: false }, // Falta +34
    { num: "+34511222333", esperado: false }, // Empieza por 5 (no válido)
    { num: "+346112223334", esperado: false } // Demasiados dígitos
];

console.table(pruebas.map(p => ({
    Número: p.num,
    Resultado: validarTelefonoEspana(p.num) ? "✅ Válido" : "❌ Inválido",
    Correcto: validarTelefonoEspana(p.num) === p.esperado ? "👍" : "👎"
})));